Documentation/project-docs/platform-names.md

```diff
@@ -0,0 +1,61 @@
+
+## .NET Core
+
+**Also referred to as**: UWP, ~~Store~~
+
+Originally, .NET Core was the identifier we used to describe the surface area Windows 8 store applications can use. When we designed the API set, we wanted to create a foundation for .NET where portability is a first class concern for the layering and componentization. For more details, read [this blog post](http://blogs.msdn.com/b/dotnet/archive/2014/12/04/introducing-net-core.aspx).
+
+Today, .NET Core no longer only powers store applications. .NET Core is the name for the open source, cross-platform stack that ASP.NET Core and UWP applications are built on. The stack includes a set of framework libraries (CoreFX), a JIT based runtime (CoreCLR), an AOT based runtime (CoreRT), and a set of tooling (such as the dotnet CLI).
+
+That's why referring to .NET Core as 'Store' is no longer correct. But you can think of today's .NET Core as an evolution of the original surface area we used for store applications. Many of the original design goals are still relevant, especially around layering and portability.
+
+## Universal Windows Platform (UWP)
+
+**Also referred to as**: Store, WinRT, Metro
+
+The Universal Windows Platform (UWP) is the platform that is used for building modern, touch-enabled Windows applications as well as headless devices for Internet of Things (IoT). It's designed to unify the different types of devices that you may want to target, including PCs, tablets, phablets, phones, and even the Xbox.
+
+UWP provides many services, such as a centralized app store, an execution environment (AppContainer), and a set of Windows runtime APIs (WinRT). While UWP itself doesn't provide .NET APIs, the Visual Studio tooling experience will present UWP as a platform that is composed of both, WinRT APIs, as well as .NET APIs. The .NET side is provided by .NET Core.
+
+## .NET Native
+
+**Also referred to as**: ahead-of-time (AOT), IL compiler (ILC)
+
+.NET Native is a compiler tool chain that will produce native code ahead-of-time (AOT), as opposed to just-in-time (JIT). The compilation can happen on the developer machine as well as on the store side, which allows blending AOT with the benefits of servicing.
+
+You can think of .NET Native as an evolution of NGEN: NGEN basically simply runs the JIT up front, the code quality and behavior is identical to the JITed version. Another downside of NGEN is that it happens on the user's machine, rather than the developer's machine. NGEN is also at the module level, i.e. for each MSIL assembly there is a corresponding NEGEN'ed assembly that contains the native code. .NET Native on the other hand is a C++ like compiler and linker. It will remove unused code, spend more time optimizing it, and produce a single, merged module that represents the closure of the application.
```

Comment:
Same here; I'll address this by adding a terms section at the top.

FYI, NGEN is often written as NGen. It simply stands for Native image generation. It's basically a persistent JIT.